src/nupic/engine/Region.cpp

```diff
@@ -512,13 +512,18 @@ bool Region::operator==(const Region &o) const {
     if (output_a->getData().getType() != output_b->getData().getType() ||
         output_a->getData().getCount() != output_b->getData().getCount())
         return false;
+    // compare output buffer contents.
+    if (output_a->getData() != output_b->getData()) return false;
     return true;
```

Comment:
nit, `return output_a->getData() == output_b->getData()`

Reply:
> nit, return output_a->getData() == output_b->getData()

When using the debugger it is nice to have a place to set the breakpoint when everything is true.